### PR TITLE
VPN-4707 - Initialize Glean on the iOS network extension

### PR DESCRIPTION
**Pull request by brizental**
~Draft as it is still missing the synchronization of uploadEnabled between netext and main app.~

This is ready for review.

Note: I suggest reviewing this commit by commit.

### CODE REVIEW

File: qtglean/ios.cmake [hunk 1]
```diff
@@ -22,12 +22,24 @@ target_include_directories(iosglean PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
...
+    BUNDLE_EXTENSION framework
+    MACOSX_BUNDLE ON
+    MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.in
+    MACOSX_BUNDLE_BUNDLE_NAME "IOSGlean"
+    MACOSX_BUNDLE_BUNDLE_VERSION "${BUILD_ID}"
+    MACOSX_BUNDLE_COPYRIGHT "MPL-2.0"
+    MACOSX_BUNDLE_GUI_IDENTIFIER "${BUILD_IOS_APP_IDENTIFIER}.iosglean"
+    MACOSX_BUNDLE_INFO_STRING "IOSGlean"
+    MACOSX_BUNDLE_LONG_VERSION_STRING "${CMAKE_PROJECT_VERSION}-${BUILD_ID}"
+    MACOSX_BUNDLE_SHORT_VERSION_STRING "${CMAKE_PROJECT_VERSION}"
```
Comment (brizental) [inline qtglean/ios.cmake:37]: Note: I think this whole Info.plist.in sitatuation is unnecessary here. I'll see if I can remove this before opening this up for review.

File: qtglean/ios.cmake [hunk 2]
```diff
@@ -22,12 +22,24 @@ target_include_directories(iosglean PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
...
     XCODE_ATTRIBUTE_SWIFT_VERSION "5.0"
     XCODE_ATTRIBUTE_CLANG_ENABLE_MODULES "YES"
     XCODE_ATTRIBUTE_SWIFT_OBJC_BRIDGING_HEADER "${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Glean.h"
     XCODE_ATTRIBUTE_SWIFT_PRECOMPILE_BRIDGING_HEADER "NO"
+    XCODE_ATTRIBUTE_PRODUCT_BUNDLE_IDENTIFIER "${BUILD_IOS_APP_IDENTIFIER}.iosglean"
```
Comment (brizental) [inline qtglean/ios.cmake:42]: Note: This is considered an invalid identifier. I need to fix it before opening up for review.

File: src/apps/vpn/cmake/ios.cmake [hunk 3]
```diff
@@ -4,18 +4,30 @@
...
 ## Install the Network Extension into the bundle.
 add_dependencies(mozillavpn networkextension)
 
-## Install the Glean iOS SDK into the bundle.
-include(${CMAKE_SOURCE_DIR}/qtglean/ios.cmake)
-add_dependencies(mozillavpn iosglean)
+# Note: Just directly using the qtglean_bindings target here didn't seem to work.
+# Instead we are just going to add the dylib directly as a framework.
+#
+# TODO: I think this is always getting the release binary.
```
Comment (brizental) [inline src/apps/vpn/cmake/ios.cmake:29]: Note to self: Look at this before merge.

File: src/apps/vpn/platforms/ios/iostunnel.swift [hunk 4]
```diff
@@ -13,12 +14,36 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
...
+        Glean.shared.registerPings(GleanMetrics.Pings.self)
+        Glean.shared.initialize(
+            uploadEnabled: telemetryEnabled,
+            configuration: Configuration.init(
+                channel: appChannel != nil ? appChannel : "unknown",
+                dataPath: gleanDataPath.relativePath
+            ),
+            buildInfo: GleanMetrics.GleanBuild.info
+        )
+    }
```
Comment (brizental) [inline src/apps/vpn/platforms/ios/iostunnel.swift:41]: Note: Add a shutdown call to `deinit` once the new Glean release comes out. It's supposed to come out tomorrow.

File: 3rdparty/glean
Comment (strseb) [inline 3rdparty/glean:1]: Is a update for glean intended?

File: src/apps/vpn/platforms/ios/iosconstants.swift [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+struct Constants {
+  static let appGroupIdentifier = "group.org.mozilla.ios.Guardian"
+
+  struct UserDefaultKeys {
+    static let telemetryEnabled = "TELEMETRY_ENABLED"
+    static let appChannel = "APP_CHANNEL"
```
Comment (strseb) [inline src/apps/vpn/platforms/ios/iosconstants.swift:10]: "APP_CHANNEL"? Should this maybe be release?

**Review by strseb — APPROVED** [review]
r+wc - Looks reasonable even though i'm not an iOS expert.

File: 3rdparty/glean
Comment (brizental) [reply 3rdparty/glean:1]: Yes, the shutdown API for iOS was made available in the latest release.

File: src/apps/vpn/platforms/ios/iosconstants.swift (hunk 5 above)
Comment (brizental) [reply src/apps/vpn/platforms/ios/iosconstants.swift:10]: This is the key, not the value.

**Pull request merged**